test/compilable/ob1.d

```diff
@@ -0,0 +1,25 @@
+
+@live int* foo1(int* p)
```

Comment:
I think comments as to what this is testing would help comprehension. IIUC, this is testing that not reading or writing to `p` despite it supposedly being an owning pointer is not an error.